src/TabContainer.js

```diff
@@ -1,138 +1,134 @@
-import React from 'react';
+import React, { useMemo } from 'react';
 import PropTypes from 'prop-types';
-import { uncontrollable } from 'uncontrollable';
+import { useUncontrolled } from 'uncontrollable';
 
 import TabContext from './TabContext';
 import SelectableContext from './SelectableContext';
 
-class TabContainer extends React.Component {
-  static propTypes = {
-    /**
-     * HTML id attribute, required if no `generateChildId` prop
-     * is specified.
-     *
-     * @type {string}
-     */
-    id(props, ...args) {
-      let error = null;
-
-      if (!props.generateChildId) {
-        error = PropTypes.string(props, ...args);
-
-        if (!error && !props.id) {
-          error = new Error(
-            'In order to properly initialize Tabs in a way that is accessible ' +
-              'to assistive technologies (such as screen readers) an `id` or a ' +
-              '`generateChildId` prop to TabContainer is required',
-          );
-        }
+/* eslint-disable react/no-unused-prop-types */
```

Comment:
@jquense bleh, is there a better way to do this? alternatively, why isn't this triggering for e.g. `<Alert>`? does this rule just not work for `React.forwardRef`s?

Reply:
hah, i guess it just doesn't work for `ForwardRef` things...

Reply:
After some investigation, it's not `forwardRef` at fault. If a component spreads props into a child, the lint rule passes. Even the following example passes:
```js
const Alert = _props => <div {...{}} />;
```

Most `forwardRef` components spread remainder props, so they already trigger this behavior.

Reply:
ah okay. well, whatever. thanks!